src/test/java/com/devonfw/sample/archunit/NamingConventionTest.java

```diff
@@ -51,6 +63,90 @@ public class NamingConventionTest {
+                   .because("Classes extending ApplicationPersistenceEntity must be located in the package dataaccess");
+
+    /**
+     * DevonAbstractUcCheck verifying that classes extending AbstractUc have to have be in layer logic
+     */
+    @ArchTest
+    private static final ArchRule DevonAbstractUcCheck = 
+            classes()
+                   .that().areAssignableTo(AbstractUc.class).should().resideInAnyPackage("..logic..")
+                   .because("Classes extending AbstractUc must be located in the package logic");
+       
+    /**
+     * DevonMapperCheck verifying that classes extending Mapper have to have be in layer logic and end with the suffix Mapper
+     */
+    @ArchTest
+    private static final ArchRule DevonMapperCheck = 
+            classes()   
+                   .that().areAnnotatedWith(Mapper.class).should().resideInAnyPackage("..logic..").andShould().haveSimpleNameEndingWith("Mapper")
+                   .because("Classes extending Mapper must be located in the package logic and end with Mapper");
+
+    /**
+     * DevonPathCheck verifying that classes extending Path have to have be in layer service and end with the suffix Service
+     */
+    @ArchTest
+    private static final ArchRule DevonPathCheck = 
+            classes()   
+                   .that().areAnnotatedWith(Path.class).should().resideInAnyPackage("..service..").andShould().haveSimpleNameEndingWith("Service")
+                   .because("Classes extending Path must be located in the package service and end with Service");
+
+    /**
+     * DevonAbstractEtoCheck verifying that classes extending AbstractEto have to have be in layer common and implement an interface with the same SimpleName excluding Eto and is in the same Package
+     */                  
+    @ArchTest
+    private static final ArchRule DevonAbstractEtoCheck = 
+            classes()
+                   .that().areAssignableTo(AbstractEto.class)
+                   .and().doNotHaveSimpleName("AbstractEto")
+                   .should().resideInAnyPackage("..common..")
+                   .andShould(new ArchCondition<JavaClass>("implemted a interface with same simple name", new Object(){}) 
+                   {
+                        @Override  
+                        public void check(JavaClass javaClass, ConditionEvents events) {
+                                String supposedInterfaceName = javaClass.getPackageName() + "." + javaClass.getSimpleName().replace("Eto", "");
+                                boolean hasCorrectInterface = javaClass.getInterfaces().stream().anyMatch(i->i.getName().equals(supposedInterfaceName));
+                                String message = "The Testresult of " + javaClass.getSimpleName() + " was " + hasCorrectInterface;
+                                events.add(new SimpleConditionEvent(javaClass, hasCorrectInterface, message));
+                        }
+                        
+                   });               
```

Comment:
I would merge this with the existing ETO rule `N4DevonNamingConventionEtoCheck`.